src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookMultibranchTrigger.java

```diff
@@ -9,34 +12,63 @@
 import jenkins.branch.MultiBranchProject;
 import org.jenkinsci.Symbol;
 import org.kohsuke.stapler.DataBoundConstructor;
+
 import javax.inject.Inject;
 import java.util.logging.Logger;
 
 public class BitbucketWebhookMultibranchTrigger extends Trigger<MultiBranchProject<?, ?>> {
 
+    //the version (of this class) where the PR trigger was introduced. Version is 0 based.
+    private static final int BUILD_ON_PULL_REQUEST_VERSION = 1;
     private static final Logger LOGGER = Logger.getLogger(BitbucketWebhookMultibranchTrigger.class.getName());
 
     private final boolean pullRequestTrigger;
     private final boolean refTrigger;
+    /**
+     * This exists as a simple upgrade task. Old classes will de-serialise this to default value (of 0). New
+     * classes will serialise the actual value that was stored. Because the constructor is not run during de-serialisation
+     * we can safely set the value in the constructor to indicate which version this class is.
+     */
+    private final int version;
 
     @SuppressWarnings("RedundantNoArgConstructor") // Required for Stapler
     @DataBoundConstructor
     public BitbucketWebhookMultibranchTrigger(boolean pullRequestTrigger, boolean refTrigger) {
         this.refTrigger = refTrigger;
         this.pullRequestTrigger = pullRequestTrigger;
+        version = BUILD_ON_PULL_REQUEST_VERSION;
     }
 
     @Override
     public BitbucketWebhookMultibranchTrigger.DescriptorImpl getDescriptor() {
         return (BitbucketWebhookMultibranchTrigger.DescriptorImpl) super.getDescriptor();
     }
 
+    /**
+     * Is the trigger applicable for the given webhook. If the trigger is configured for RefChange and the event
+     * is a PR opened event it should return false. The trigger should <em>NOT</em> trigger as a result of this call.
+     *
+     * @param event the webhook as it was received
+     * @return true if this trigger is applicable to the given webhook
+     * @since 3.0.0
+     */
+    public boolean isApplicableForEventType(AbstractWebhookEvent event) {
```

Comment:
Having looked at this in full context, I will advocate _for_ this approach. Looking at core jenkins implementations of the trigger, `ReverseBuildTrigger.java` has a `shouldTrigger` method with a similar purpose, though it is invoked quite differently. In all cases I can see, Jenkins only ever expressly forbids a trigger running if the job is null.
Besides consistency, the presence of the method I think is good documentation on how the trigger should be consumed, rather than opaquely performing that check and silently failing if the event is wrong.

Reply:
Thank you, as per our discussion I'm still on the fence about it, but I do like the clearer intention this offers.